sources/api/prairiedog/src/main.rs

```diff
@@ -277,6 +283,20 @@ fn load_crash_kernel() -> Result<()> {
     Ok(())
 }
 
+async fn reboot_if_required<P>(socket_path: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    if is_reboot_required(socket_path).await? {
+        info!("Boot settings changed and require a reboot to take effect. Initiating reboot...");
+        command("/usr/bin/systemctl", &["reboot"])?;
```

Comment:
Did we lose the property from the previous revision where the system would never continue starting other units?

The manual for `systemctl reboot` says it's mostly equivalent to:
```
systemctl start reboot.target --job-mode=replace-irreversibly --no-block
```

It might be good to somehow confirm that `activate-multi-user.target` never gets a chance to run. I'd guess that since it has `DefaultDependencies=yes` (implied) then it will conflict with `shutdown.target` and never actually start because the stop job will run first.